### ISSUE TITLE
Update accessibility checker documentation examples to enable more relevant rules

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#12188]
### Pertinent section of the Wagtail docs

[Built-in accessibility checker](https://docs.wagtail.org/en/stable/advanced_topics/accessibility_considerations.html#built-in-accessibility-checker) docs and the [bakerydemo wagtail_hooks](https://github.com/wagtail/bakerydemo/blob/main/bakerydemo/base/wagtail_hooks.py).

### Details

Follow-up to #12157. In the docs, we’re showing how site implementers can enable way more rules than the checker UI in the CMS has been designed for. This isn’t a good experience for sites where content management users aren’t developers or don’t have deep accessibility expertise, as a lot of the errors will be about aspects of the site that are unrelated to the CMS.

Instead, we should review [available Axe rules](https://github.com/dequelabs/axe-core/blob/develop/doc/rule-descriptions.md) and in the docs / bakerydemo:

1. Only enable rules that are at least tangentially related to content considerations.
2. Also demonstrate how to override the rules’ messages. For example "Contact the project’s developer to investigate this issue with specialist tools".
3. Add an explanation to the docs that though enabling all rules is possible, it should only be done for sites where there are few to no issues expected and any CMS users would be able to know what to do with any unforeseen issues.

### Working on this

Assigning to @albinazs

**Comment by thibaudcolas** [comment]
```python
class CustomAccessibilityItem(AccessibilityItem):
    axe_custom_checks = [
        {
            "id": "p-as-heading",
            "options": {
                "margins": [
                    {
                        "weight": 100,
                        # "italic": True,
                    },
                ],
                "passLength": 1,
                "failLength": 0.5
            },
        },
    ]
```
